linkerd/http/upgrade/src/upgrade.rs

```diff
@@ -39,7 +39,7 @@ struct Http11UpgradeHalves {
 
 /// A marker type inserted into Extensions to signal it was an HTTP CONNECT
 /// request.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct HttpConnect;
```

Comment:
I wonder if we should be a little more defensive about cloning this. What does cloning it _mean_? When handling a CONNECT request, we can't clone the request, really. (Technically, we can't clone the body, but practically, it means we can't clone the request). Can we easily track whether this was accidentally cloned (i.e. with a custom Clone impl or Arc or some such) and validate at runtime (i.e., in proxy::http::h1) that everything is copacetic?

Reply:
that seems like a sensible idea! we did the same for its sibling, `Http11Upgrade`, in https://github.com/linkerd/linkerd2-proxy/pull/3540.

i'll add a custom `Clone` impl so that we can defend against accidental clones of this marker.

Reply:
i've opened #3779 to remove this extension altogether. we only use it in one place, and that code is simplified if we do not rely on response extensions to signal upgrades.